frontend/packages/ceph-storage-plugin/src/components/modals/delete-snapshot/delete-snapshot.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+
+import { HandlePromiseProps, withHandlePromise } from '@console/internal/components/utils';
+import {
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+  createModalLauncher,
+} from '@console/internal/components/factory';
+
+import { Form } from '@patternfly/react-core';
+import { VolumeSnapshotModel } from '../../../models';
+import { k8sKill } from '@console/internal/module/k8s';
+
+export const DeleteSnapshotModal = withHandlePromise((props: DeleteSnapshotModalProps) => {
+  const { close, cancel, resource, errorMessage, inProgress } = props;
+
+  const submit = (event: React.FormEvent<EventTarget>) => {
+    event.preventDefault();
+    props
+      .handlePromise(k8sKill(VolumeSnapshotModel, resource))
+      .then(() => {
+        close();
+      })
+      .catch((error) => {
+        throw error;
+      });
+  };
+
+  return (
+    <Form onSubmit={submit}>
+      <div className="modal-content modal-content--no-inner-scroll">
+        <ModalTitle>Delete Snapshot</ModalTitle>
+        <ModalBody className="modal-body">
+          <div className="co-delete-modal">
+            <div>
+              <div>
+                Are you sure you want to delete{' '}
+                <strong className="co-break-word">{resource.metadata.name} ?</strong>
+              </div>
+            </div>
+          </div>
+        </ModalBody>
+        <ModalSubmitFooter
+          errorMessage={errorMessage}
+          inProgress={inProgress}
+          submitDanger
+          submitText="Delete"
+          cancel={cancel}
+        />
+      </div>
+    </Form>
+  );
+});
+
+export type DeleteSnapshotModalProps = {
+  resource?: any;
```

Comment:
type of `Resource` I am assuming its K8sResourceKind, but if K8sResourceCommon suffices, use that.

Reply:
```suggestion
  resource?: K8sResourceCommon;
```

Is it really optional?

Reply:
ack